### ISSUE TITLE
Text displayed incorrectly after update to 0.70 [bug]

### ISSUE THREAD

**Issue by selivan** [zim-desktop-wiki/zim-desktop-wiki#727]
Environment:

OS: Ubuntu 18.04 Bionic LTS
DE: Unity with usual Compiz as DM
Python: 3.6.7
GTK3: 3.22.30-1ubuntu2
GTK2: 2.24.32-1ubuntu1

Steps to reproduce:

- open zim with any notebook
- try scrolling text down or switching to other page
- try switching to other windows and then back to zim

Result: text is displayed incorrectly. On scrolling or switching to other page text is completely distroyed. After switching to other windows and back their content is displayed in text area of zim window.

Expected result: text displayed correctly.

Notes: zim 0.68 from Ubuntu repository works fine.

![zim-0 7 0-bug-text-2019-03-30-15-19-34](https://user-images.githubusercontent.com/1208989/55276041-50f6cb00-52ff-11e9-8b78-13747a85b37a.png)

**Comment by jaap-karssenberg** [comment]
It looks like the window is not properly re-drawn. Probably this is a communication issue in the interaction between Gtk3 and the window manager. Not sure if/how to force the re-draw from zim.

Might have similar root cause as #642, but could also be unrelated

**Comment by jaap-karssenberg** [comment]
Don't think this one is related to the other rendering issues. Also seems no other reports so far. Anyone seen this under other environments ?

**Comment by selivan** [comment]
@waldauf , you replied with :+1: , could you please write if you have also seen this bug?

**Comment by waldauf** [comment]
@selivan Yep, I also have this problem with v0.70-2. That's why I had to downgrade to v0.69. 

I tried repository version and download from github. In both cases I have problem with incorrectly displayed text. But just if I have in text some Source code. Because after upgrade Source code plugin stopped work (it is written in the source code border header):
![image](https://user-images.githubusercontent.com/24603883/56083058-77008d00-5e20-11e9-9c23-764a488fde66.png)

I tried to turn on Source code plugin. But in plugin info is written: _GtkSourceView - unresolved_.  I tried to install newer version of GtkSourceView - `gtksourceview3-3.24.10-1`. Then is possible to turn on Source code plugin, but text is still incorrectly displayed:
![image](https://user-images.githubusercontent.com/24603883/56083078-dced1480-5e20-11e9-9da9-5bbc294979cb.png)
... every time when I open this page is text chaotically  hidden.

In my repository I also have package gtksourceview4, but it doesn't work (the Source code plugin cannot be enabled).

If you need more information write me. I would like if I can help you to catch this bug.

 System information:
- OS: `Arch linux`
- GTK2: `extra/gtk2 2.24.32-1`
- GTK3: `extra/gtk3 1:3.24.8-1`
- GTK Sourceview: `extra/gtksourceview2 2.10.5-4`
- GTK Sourceview: `gtksourceview3-3.24.10-11`

The page from v0.69 without any problem:
![image](https://user-images.githubusercontent.com/24603883/56083162-a49a0600-5e21-11e9-98cc-716ffe1a3f16.png)

**Comment by vossenjp** [comment]
I *think* I am seeing this also, but only sometimes.  Most of the pages I'm on constantly are >2,000 lines, but I've just started a new one that's only <150.  That new one is often garbled when I switch to it, but a CTRL-R fixes it.  It almost seems like there is a cache sync problem.  And I can not reliably reproduce it, it just happens sometimes.  I do not recall ever having similar problems on 0.69.1 or lower.

**Comment by jaap-karssenberg** [comment]
@waldauf the issue you show is different, please see #642

**Comment by waldauf** [comment]
@jaap-karssenberg I'm sorry for my mistake. I didn't know about this issue.
